crates/ol/stf/src/transaction_processing.rs

```diff
@@ -99,71 +122,65 @@ fn process_update_tx<S: IStateAccessor>(
+        let coin = astate
+            .take_balance(total_sent)
+            .map_err(|_| ExecError::InsufficientAccountBalance(target, total_sent))?;
+        coin.safely_consume_unchecked(); // TODO: better usage?
+
+        // Now get snark account state and update proof state
         let sastate = astate
             .as_snark_account_mut()
             .map_err(|_| ExecError::IncorrectTxTargetType)?;
 
-        // 2. Call the snark account machinery to process the update.
-        //
-        // XXX This implementation is very limited because we don't want to support
-        // the full snark account functionality yet.  We don't check anything, we
-        // just update the fields as we're told to without authenticating anything.
-        //
-        // TODO make this the full implementation, this is where we'd call out to it
-        // instead of just doing it here
-
-        // This just calls the function to update the state as we would if we
-        // actually were doing the real implementation.
         sastate.update_inner_state(
-            new_state.inner_state(),
-            new_state.next_inbox_msg_idx(),
-            seqno.into(),
-            extra_data,
+            operation.new_state().inner_state(),
+            operation.new_state().next_inbox_msg_idx(),
+            operation.seq_no().into(),
+            operation.extra_data(),
         )?;
 
-        // We also have to extract the effects here too, subtracting balance in the
-        // process.
+        // Collect effects using snark-acct-sys
         let mut fx_buf = AcctInteractionBuffer::new_empty();
-        for m in outputs.messages() {
-            let coin = astate
-                .take_balance(m.payload().value())
-                .map_err(|_| ExecError::InsufficientAccountBalance(target, m.payload().value()))?;
-            coin.safely_consume_unchecked(); // TODO track this better
-            fx_buf.send_message_to(m.dest(), m.payload().clone());
-        }
-
-        // Capture the account serial for the log.
-        let account_serial = astate.serial();
+        snark_sys::apply_update_outputs(&mut fx_buf, verified_update)?;
 
-        Ok((fx_buf, account_serial))
+        Ok(fx_buf)
     })??;
 
-    // 3. Apply the effects (after the account update closure completes).
+    // Step 4: Apply effects
     apply_interactions(state, target, fx_buf, context.basic_context())?;
 
-    // 4. Emit a log message.
-    // According to the spec, the log should contain:
-    // - new_msg_idx: The sequence number from the account state
-    // - extra_data: The extra data from the update operation
-    // TODO improve codec error handling here when more stuff is SSZed
-    let log_data =
-        SnarkAccountUpdateLogData::new(new_state.next_inbox_msg_idx, extra_data.to_vec()).ok_or(
-            ExecError::Codec(strata_codec::CodecError::OverflowContainer),
-        )?;
+    // Step 5: Emit log
+    let log_data = SnarkAccountUpdateLogData::new(
```

Comment:
what is this log for? declaring the next msg idx? is there anything we (anyone\thing) do with this log?

Reply:
No.. this log is not needed and should not be emitted. Good catch!

Reply:
This might have been an oversight or something from an earlier design framing.  At some point the logs would have been supplementary in making state reconstruction work, but this was not fully thought-through and a bad idea anyways.

Since now in DA the plan is explicitly to record the `extra_data` in each call to update the account's state, this is probably actually purely redundant.